downloadservice/app.py

```diff
@@ -312,8 +317,10 @@ def generate(stats):
                         len(r)/1024**2, stats['total_written']/1024**2)
             stats['total_written'] += len(r)
             yield r
+        yield r
 
-    r = upstream_session.put(url, data=generate(stats))
+    # r = upstream_session.put(url, data=generate(stats))
```

Comment:
yes but we really want stream. This will just read all at once, right? It was first solution of course, but the outcome was trying to `request.stream.read()` a 100Gb file and crash the hub. It should only be in memory in managable chunks.

If test does not work, maybe webdav server of the fixture does not support streaming like that?..